catalog/src/main/kotlin/com/adevinta/spark/catalog/datastore/buttonconfigurator/ButtonsConfiguratorProperties.kt

```diff
@@ -0,0 +1,48 @@
+/*
+ * Copyright (c) 2023 Adevinta
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+package com.adevinta.spark.catalog.datastore.buttonconfigurator
+
+import androidx.core.content.res.ResourcesCompat
+import kotlinx.serialization.Serializable
+
+@Serializable
+internal data class ButtonsConfiguratorProperties(
+    val isEnabled: Boolean,
+    val isLoading: Boolean,
+    val iconSide: String,
+    val intent: String,
+    val size: String,
+    val style: String,
+    val iconId: Int,
+) {
+    companion object {
+        val DEFAULT = ButtonsConfiguratorProperties(
+            isEnabled = true,
+            isLoading = false,
+            iconSide = "START",
```

Comment:
Shouldn't we use the actual enums here?

Reply:
(Same for other \*Properties) I wanted to avoid dependency between model from `com.adevinta.spark.components.*` and Datastore model